sdk/core/System.ClientModel/src/Pipeline/ApiKeyAuthenticationPolicy.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Primitives;
+
+public class ApiKeyAuthenticationPolicy : PipelinePolicy
+{
+    private readonly string _name;
+    private readonly string? _keyPrefix;
+    private readonly KeyLocation _location;
+    private readonly ApiKeyCredential _credential;
+
+    /// <summary>
+    /// Create a new instance of the <see cref="ApiKeyAuthenticationPolicy"/> class, where the
+    /// credential value will be specified in a request header.
+    /// </summary>
+    /// <param name="credential">The <see cref="ApiKeyCredential"/> used to authenticate requests.</param>
+    /// <param name="headerName">The name of the request header used to send the key credential in the request.</param>
+    /// <param name="keyPrefix">A prefix to prepend before the key credential in the header value.
+    /// If provided, the prefix string will be followed by a space and then the credential string.
+    /// For example, setting <c>valuePrefix</c> to "SharedAccessKey" will result in the header value
+    /// being set fo "SharedAccessKey {credential.Key}".</param>
+    public static ApiKeyAuthenticationPolicy CreateHeaderApiKeyPolicy(ApiKeyCredential credential, string headerName, string? keyPrefix = null)
+    {
+        Argument.AssertNotNull(credential, nameof(credential));
+        Argument.AssertNotNullOrEmpty(headerName, nameof(headerName));
+
+        return new ApiKeyAuthenticationPolicy(credential, headerName, KeyLocation.Header, keyPrefix);
+    }
+
+    public static ApiKeyAuthenticationPolicy CreateQueryApiKeyPolicy(ApiKeyCredential credential, string queryName)
+    {
+        // TODO: Add tests for this implementation if the API is approved.
```

Comment:
Is this comment still relevant?

Reply:
Not if we cut the feature :)  I'll remove it either way